src/write.rs

```diff
@@ -143,13 +219,11 @@ mod test {
     fn test_writer() -> io::Result<()> {
         {
             let data = b"0123456789ABCDEF";
-            let mut writer = super::BGzWriter::new(fs::File::create("tmp/test.gz")?);
-
+            let mut writer = super::BGzWriter::new(fs::File::create("tmp/test.gz")?, 3);
             for _ in 0..30000 {
                 writer.write(&data[..])?;
             }
```

Comment:
The chain of events to shutdown things in the face of an IO error looks complicated, so it would be good to have some tests that exercise that.

Reply:
Agreed. Perhaps a custom writer class that dies after 30B of written material. Hmm